template/flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  description = "An example of Napalm with flakes";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/master";
+
+  # Import napalm
+  inputs.napalm.url = "github:nix-community/napalm";
+
+  outputs = { self, nixpkgs, napalm }:
+    let
+      # Generate a user-friendly version numer.
+      version = builtins.substring 0 8 self.lastModifiedDate;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          # Use napalm as your overlay
+          overlays = [ self.overlay napalm.overlay ];
+        });
+
+    in {
+
+      # A Nixpkgs overlay.
+      overlay = final: prev:
+        let
+          # In case you do not want to use overlay, you can use:
+          buildNapalmPackage = (napalm.overlay final prev).napalm.buildPackage;
```

Comment:
I would say this is advanced stuff and people who care about this can trivially recreate it so I would not include it in the template.